### ISSUE TITLE
Matcher returning object in old/incorrect format?

### ISSUE THREAD

**Issue by idolize** [unindented/custom-immutable-matchers#43]
I'm seeing the following error when using `.toEqualImmutable`:

```
Unexpected return from a matcher function.
Matcher functions should return an object in the following format:
      {message?: string | function, pass: boolean}
'{"compare": [Function anonymous]}' was returned
```

![Screen Shot 2021-07-08 at 2 48 02 PM](https://user-images.githubusercontent.com/635005/124975212-ca5fe680-dffb-11eb-8602-6f9e8ae289e3.png)

**Comment by unindented** [comment]
@idolize I no longer use this library, but I'm open to merging PRs if you figure out what the issue is.

**Comment by dimitriylol** [comment]
@unindented thanks for the quick merge. Can you publish the new package version into npm? I'm not sure I can do it.

**Comment by unindented** [comment]
@dimitriylol sorry for the delay. I've now published `v3.0.0`. Thanks all!

**Comment by idolize** [comment]
Thanks @dimitriylol and @unindented!
